shared/directives/drag-scroll/drag-scroll.directive.ts

```diff
@@ -0,0 +1,66 @@
+import { Directive, ElementRef, OnInit } from '@angular/core';
+import { fromEvent } from 'rxjs';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+
+@UntilDestroy()
+@Directive({
+  selector: '[appDragScroll]',
+})
+export class DragScrollDirective implements OnInit {
+  private dragScrollElement: HTMLElement;
+  private isMovable: boolean;
+  private lastEvent: MouseEvent;
+
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+  ) {
+    this.dragScrollElement = elementRef.nativeElement;
+  }
+
+  ngOnInit(): void {
+    fromEvent(this.dragScrollElement, 'mouseover').pipe(
+      untilDestroyed(this),
+    ).subscribe(() => {
+      if (this.dragScrollElement.scrollWidth > window.innerWidth) {
```

Comment:
This code will work in our case coz the navigation bar has the same visible width as the window. Any other element will be broken
`> element.offsetWidth`